docs/root/start/quick-start/securing.rst

```diff
@@ -80,27 +80,34 @@ Firstly, you can ensure that the certificates are from a mutually trusted certif
    :language: yaml
    :linenos:
    :lineno-start: 42
-   :lines: 42-52
-   :emphasize-lines: 8-11
+   :lines: 42-53
+   :emphasize-lines: 9-12
    :caption: :download:`envoy-demo-tls-validation.yaml <_include/envoy-demo-tls-validation.yaml>`
 
-You can also ensure that the "Subject Alternative Names" for the cerficate match.
+You should also ensure that the Subject Alternative Names (SANs) for the certificate match.
 
 This is commonly used by web certificates (X.509) to identify the domain or domains that a
 certificate is valid for.
 
+For the most common case where the certificate should have a SAN matching the :ref:`SNI <start_quick_start_securing_sni_client>`
+which was sent, you can enable :ref:`auto_sni_san_validation <envoy_v3_api_field_extensions.transport_sockets.tls.v3.UpstreamTlsContext.auto_sni_san_validation>`
+and omit :ref:`match_typed_subject_alt_names <envoy_v3_api_field_extensions.transport_sockets.tls.v3.CertificateValidationContext.match_typed_subject_alt_names>`
+in the validation context.
+
+To validate that a certificate has a SAN matching the downstream request ``host`` or ``:authority`` header, you can enable
+:ref:`auto_san_validation <envoy_v3_api_field_config.core.v3.UpstreamHttpProtocolOptions.auto_san_validation>`.
+
+When multiple validation options are configured, ``auto_san_validation`` has the highest priority, followed by ``auto_sni_san_validation``,
+followed by ``match_typed_subject_alt_names``.
+
 .. literalinclude:: _include/envoy-demo-tls-validation.yaml
    :language: yaml
    :linenos:
    :lineno-start: 44
-   :lines: 44-54
-   :emphasize-lines: 6-7, 10-11
+   :lines: 44-55
+   :emphasize-lines: 6-7, 11-12
    :caption: :download:`envoy-demo-tls-validation.yaml <_include/envoy-demo-tls-validation.yaml>`
 
-.. note::
-
-   If the "Subject Alternative Names" for a certificate are for a wildcard domain, eg ``*.example.com``,
```

Comment:
This note didn't make sense; Envoy will correctly validate an expected SAN of `test.example.com` against a certificate with `*.example.com` (that is considered a match)

Reply:
it was quite a while when i added this - but i thought i added it after testing it out

Reply:
It's documented in `match_typed_subject_alt_names`; this definitely works

Reply:
hmm - not sure - but im wondering if there are some circumstances where it doesnt work

Reply:
If there are, it's a bug and we should fix it.

Reply:
if i get some time ill try and retry the steps i might have taken